sql/core/src/main/scala/org/apache/spark/sql/expressions/UserDefinedFunction.scala

```diff
@@ -41,12 +41,16 @@ import org.apache.spark.sql.types.DataType
 case class UserDefinedFunction protected[sql] (
     f: AnyRef,
     dataType: DataType,
-    inputTypes: Option[Seq[ScalaReflection.Schema]]) {
+    inputTypes: Option[Seq[DataType]]) {
```

Comment:
+1. This is why we added _nameOption, _nullable and _deterministic in 2.3 release. 

Please also remove the changes of MimaExcludes.scala made in https://github.com/apache/spark/pull/22259